node_modules/oae-core/setpermissions/js/setpermissions.js

```diff
@@ -115,12 +120,39 @@ define(['jquery', 'oae.core'], function($, oae) {
             });
         };
 
+        /**
+         * Get the members of the folder if the resource is inside of a folder
+         *
+         * @param  {String}         folderId                    The folder id
+         * @param  {Function}       callback                    Standard callback function
+         */
+        var getFolderMembers = function(folderId, callback) {
+
+            if (oae.data.me.id === folderId) {
+                return callback(null, []);
+            }
+
+            oae.api.folder.getMembers(folderId, null, null, function(err, members) {
+                if (err) {
+                    return callback(err);
+                };
+
+                var membersProfile = _.map(members.results, function(member) { return member.profile; });
+                folderMembers = _.reject(membersProfile, function(profile) { return profile.id === oae.data.me.id; });
+
+                return callback();
```

Comment:
instead of assigning a variable what is lost somewhere above, why not return it via callback? This way we avoid having "state" and we just pass arguments to functions accordingly.